src/v/kafka/server/handlers/fetch.cc

```diff
@@ -1336,12 +1336,20 @@ class simple_fetch_planner final : public fetch_planner::impl {
               }
 
               auto& tp = fp.topic_partition;
+              auto tn_view = tp.as_tn_view();
+              const auto& metadata_cache = octx.rctx.metadata_cache();
+              auto partition_id = tp.get_partition();
 
-              if (unlikely(octx.rctx.metadata_cache().is_disabled(
-                    tp.as_tn_view(), tp.get_partition()))) {
+              if (unlikely(metadata_cache.is_disabled(tn_view, partition_id))) {
                   resp_it->set(make_partition_response_error(
-                    fp.topic_partition.get_partition(),
-                    error_code::replica_not_available));
+                    partition_id, error_code::replica_not_available));
+                  ++resp_it;
+                  return;
+              }
+
+              if (unlikely(metadata_cache.should_reject_reads(tn_view))) {
+                  resp_it->set(make_partition_response_error(
+                    partition_id, error_code::invalid_topic_exception));
```

Comment:
question for future: what error should we return in the case of disabled read/writes

Reply:
IDK, I'd find it confusing if we returned `unknown_topic_or_partition` as the topic exists and will be shown in the topic list. Do you have any considerations?